src/api/result_proxy_api_client.rs

```diff
@@ -0,0 +1,324 @@
+    /// 1. Authentication and authorization validation
+    /// 2. Result data validation (signatures, hashes, etc.)
+    /// 3. IPFS upload and pinning
+    /// 4. Registration of the result link on the blockchain
+    ///
+    /// # Arguments
+    ///
+    /// * `authorization` - The bearer token for authenticating with the result proxy
+    /// * `result_model` - The [`ResultModel`] containing the computation result to upload
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(String)` - The IPFS link where the result was uploaded (e.g., "ipfs://QmHash...")
+    /// * `Err(reqwest::Error)` - HTTP client error or server-side error
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error in the following situations:
+    /// * Network connectivity issues preventing the HTTP request
+    /// * Authentication failures (invalid or expired token)
+    /// * Server-side validation failures (invalid signatures, malformed data)
+    /// * IPFS upload failures on the result proxy side
+    /// * HTTP status codes indicating server errors (4xx, 5xx)
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use crate::api::result_proxy_api_client::{ResultProxyApiClient, ResultModel};
+    ///
+    /// let client = ResultProxyApiClient::new("https://result-proxy.iex.ec");
+    /// let result_model = ResultModel {
+    ///     chain_task_id: String::from("0x123..."),
+    ///     zip: compressed_data,
+    ///     deterministic_hash: computed_hash,
+    ///     enclave_signature: tee_signature,
+    ///     ..Default::default()
+    /// };
+    ///
+    /// match client.upload_to_ipfs("Bearer token123", &result_model) {
+    ///     Ok(ipfs_link) => {
+    ///         println!("Successfully uploaded to: {}", ipfs_link);
+    ///         // IPFS link can be used to retrieve the result later
+    ///     }
+    ///     Err(e) => {
+    ///         eprintln!("Upload failed: {}", e);
+    ///         // Handle error appropriately (retry, report, etc.)
+    ///     }
+    /// }
+    /// ```
+    pub fn upload_to_ipfs(
+        &self,
+        authorization: &str,
+        result_model: &ResultModel,
+    ) -> Result<String, reqwest::Error> {
+        let url = format!("{}/v1/results", self.base_url);
+        let response = self
+            .client
+            .post(&url)
+            .header("Authorization", authorization)
+            .json(result_model)
+            .send()?;
+
+        if response.status().is_success() {
+            response.text()
+        } else {
+            Err(response.error_for_status().unwrap_err())
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+    use wiremock::{
+        Mock, MockServer, ResponseTemplate,
+        matchers::{body_json, header, method, path},
+    };
+
+    // Test constants
+    const TEST_TASK_ID: &str = "0x123";
+    const TEST_DEAL_ID: &str = "0x456";
+    const TEST_DETERMINISTIC_HASH: &str = "0xabc";
```

Comment:
To stay compliant with ResultModel field name from Java implementation
```suggestion
    const TEST_DETERMINIST_HASH: &str = "0xabc";
```